syne_tune/optimizer/schedulers/searchers/random_grid_searcher.py

```diff
@@ -329,22 +329,21 @@ def _next_candidate_on_grid(self) -> Optional[dict]:
         """
 
         num_combinations = len(self.hp_values_combinations)
-        if self._next_index < num_combinations:
-            is_not_done = True
-            while is_not_done and self._next_index < num_combinations:
-                candidate = dict(
-                    zip(self.hp_keys, self.hp_values_combinations[self._next_index])
-                )
-                self._next_index += 1
-                is_not_done = self._all_initial_configs.contains(candidate)
-            if self._allow_duplicates and self._next_index == num_combinations:
-                # Another round through the grid
-                self._next_index = 0
-                self._all_initial_configs = ExclusionList.empty_list(self._hp_ranges)
-            return candidate
-        else:
-            # No more candidates
-            return None
+        candidate = None
+        is_not_done = True
```

Comment:
Two suggestions:
1. `done` is unclear, what does it mean here? Recommend making it more specific, such as `is_search_space_exhausted` (or whatever it's actually checking for)
2. Recommend avoiding the negation in the variable name, as I generally find positive boolean variables more readable, and then negate it using the Python syntax of `not`.

Reply:
Actually, I can get rid of this flag altogether